s3fs/core.py

```diff
@@ -31,6 +31,14 @@
 
 _VALID_FILE_MODES = {'r', 'w', 'a', 'rb', 'wb', 'ab'}
 
+S3_SCHEMES = ['s3://', 's3a://', 's3n://']
```

Comment:
You can instead set the S3FileSystem protocol attribute and use the `._strip_protocol()` class method. This nicely removes duplicate code which you have found in this PR.

(@TomAugspurger , did the change to a `.protocols` property go in?)

Reply:
OK.
I Get the idea. Change .protocol to a list. Indeed it would be better. It would also permit to support `viewfs` for example which is currently missing.
Currently this is not merged in https://github.com/intake/filesystem_spec  and there is no PR.
Are you currently working on that ? Otherwise I might do it. Seems pretty simple.

Reply:
One question about the code factorization.

What about `split_path` function ? The best would be to move this inside S3FileSystem and also use ._strip_protocol but this would break backward compatibility.

Reply:
I thought @TomAugspurger has written this code somewhere already; I am not working on it.

`split_path` is a little different, since it cares about bucket / key, but principled no opinion on whether it is  a function or method of S3FileSystem. It would make sense for it to make use of `._strip_protocol`, though.

Reply:
Thanks for the feedback.
Turns out setting a list is actually working on `.protocol` (I was looking for the  `.protocols` property before). I pushed the review again. Also removed s3n as I'm not using this and it is deprecated.